pkg/watermark/wmb/idle_manager.go

```diff
@@ -36,11 +36,13 @@ func NewIdleManager(length int) *IdleManager {
 	}
 }
 
-// Exists returns true if the given toBuffer partition name exists in the IdleManager map.
-func (im *IdleManager) Exists(toBufferPartitionName string) bool {
+// NeedToSendCtrlMsg returns true if the given partition hasn't got any control message and needs to create a new control message
+func (im *IdleManager) NeedToSendCtrlMsg(toBufferPartitionName string) bool {
 	im.lock.RLock()
```

Comment:
make `IdleManager` private? also return interface when `NewIdleManager` is called?